src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -293,21 +308,34 @@ def update_existing_check_in(
         )
         return
 
-    # update date_added for heartbeat
-    date_updated = existing_check_in.date_updated
-    if updated_status == CheckInStatus.IN_PROGRESS:
-        date_updated = start_time
-
-    updated_timeout_at = get_new_timeout_at(existing_check_in, updated_status, start_time)
+    updated_checkin = {
+        "status": updated_status,
+        "duration": updated_duration,
+    }
 
-    existing_check_in.update(
-        status=updated_status,
-        duration=updated_duration,
-        date_updated=date_updated,
-        timeout_at=updated_timeout_at,
+    # XXX(epurkhiser): We currently allow a existing timed-out check-in to
+    # have it's duration updated. This helps users understand that a check-in
+    # DID complete. However we will NOT currently transition the status away
+    # from TIMEOUT.
+    #
+    # In the future we will likely revisit this by adding as `substatus` to
+    # check-ins which can help in the scenario where a TIMEOUT check-in
+    # transitions to a USER_TERMINAL_VALUES late value.
+    if existing_check_in.status == CheckInStatus.TIMEOUT:
+        del updated_checkin["status"]
+
+    # IN_PROGRESS heartbeats bump the timeout
+    updated_checkin["timeout_at"] = get_new_timeout_at(
+        existing_check_in,
+        updated_status,
+        start_time,
     )
 
-    return
+    # IN_PROGRESS heartbeats bump the date_updated
+    if updated_status == CheckInStatus.IN_PROGRESS:
+        updated_checkin["date_updated"] = start_time
```

Comment:
worth adding a test to cover this case?

Reply:
I'll follow up -- we've had this behavior for a while, this change doesn't actually change this behavior, so it was already uncovered.